runtime/permissions.rs

```diff
@@ -234,24 +251,32 @@ impl UnaryPermission<ReadDescriptor> {
     self.query(path)
   }
 
-  pub fn check(&self, path: &Path) -> Result<(), AnyError> {
+  pub fn check(&mut self, path: &Path) -> Result<(), AnyError> {
     let (resolved_path, display_path) = resolved_and_display_path(path);
-    self
-      .query(Some(&resolved_path))
-      .check(self.name, Some(&format!("\"{}\"", display_path.display())))
+    self.query(Some(&resolved_path)).check(
+      self.name,
+      Some(&format!("\"{}\"", display_path.display())),
+      self.prompt,
+    )?;
+    self.granted_list.insert(ReadDescriptor(resolved_path));
```

Comment:
```suggestion
    let result = self.query(Some(&resolved_path)).check(
      self.name,
      Some(&format!("\"{}\"", display_path.display())),
      self.prompt,
    )?;
    if self.prompt {
      if result.is_ok() {
        self.granted_list.insert(ReadDescriptor(resolved_path));
      } else {
        self.denied_list.insert(ReadDescriptor(resolved_path));
      }
    }
    result
```
Ditto for the other kinds and similar for `UnitPermission`.